assets/src/edit-story/app/story/useStoryReducer/reducers/utils.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * Internal dependencies
+ */
+import { LAYER_DIRECTIONS } from '../../../../constants';
+
+export function intersect( first, ...rest ) {
+	if ( ! first || ! rest || rest.length === 0 ) {
+		return first;
+	}
+
+	return rest.reduce(
+		( intersection, list ) =>
+			intersection.filter( ( value ) => list.includes( value ) ),
+		first,
+	);
+}
+
+export function isInsideRange( index, start, end ) {
+	return index >= start && index <= end;
```

Comment:
Should this ensure that `index` is numeric/a number?

Reply:
It'll return `NaN` for most thing, which is the only reasonable return value anyway.